app/src/main/java/com/spendster/presentation/WelcomeActivity.java

```diff
@@ -0,0 +1,130 @@
+package com.spendster.presentation;
+
+import android.support.v4.app.Fragment;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+
+import com.spendster.R;
+
+import java.util.ArrayList;
+
+public class WelcomeActivity extends AppCompatActivity {
+
+    private ViewPager viewPager;
+    private Button btnSkip, btnNext;
+    private Button btnSignUp, btnLogin;
+    private ArrayList<Fragment> slideFragments;
+    private ImageView[] dots;
+    private View dotsLayout;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_welcome);
+
+        initWelcomeActivity();
```

Comment:
@SimonMaha extract all code after this line from this function into "initUI()" method